### PR TITLE
[logging][rfc] add RAY_LOG_EVERY_N and RAY_LOG_EVERY_MS

### PR DESCRIPTION
**Pull request by scv119**
implement RAY_LOG_EVERY_N/RAY_LOG_EVERY_MS that logs every n entries/every n ms from the same RAY_LOG_EVERY_* line, copied from to glog's LOG_EVERY_N/LOG_EVERY_T.

TODO:
- [x] we need add glog License to https://github.com/ray-project/ray/blob/master/LICENSE

### CODE REVIEW

File: src/ray/util/logging.h [hunk 1]
```diff
@@ -87,6 +88,18 @@ enum class RayLogLevel {
...
 
+/// Macros for RAY_LOG_EVERY_N
+#define RAY_LOG_EVERY_N_VARNAME(base, line) RAY_LOG_EVERY_N_VARNAME_CONCAT(base, line)
+#define RAY_LOG_EVERY_N_VARNAME_CONCAT(base, line) base##line
+
+#define RAY_LOG_OCCURRENCES RAY_LOG_EVERY_N_VARNAME(occurrences_, __LINE__)
+
+#define RAY_LOG_EVERY_N(level, n)                             \
+  static std::atomic<uint64_t> RAY_LOG_OCCURRENCES(0);        \
+  if (ray::RayLog::IsLevelEnabled(ray::RayLogLevel::level) && \
```
Comment (ericl) [inline src/ray/util/logging.h:99]: It looks good, thou I'm wondering if we can delegate to glog under the hood which presumably already supports this

Comment (scv119) [reply src/ray/util/logging.h:99]: yup. this basically copied over what GLOG is doing: https://github.com/google/glog/blob/master/src/glog/logging.h.in#L985

**Comment by scv119** [comment]
Do we use other log backends like SPDLOG? If we only use Google Log it makes sense to just make RAY_LOG an alias of GLog.

File: src/ray/util/logging_test.cc [hunk 2]
```diff
@@ -55,6 +59,51 @@ TEST(PrintLogTest, LogTestWithoutInit) {
...
+  }
+  std::string output = GetCapturedStderr();
+  size_t occurrences = 0;
+  std::string::size_type start = 0;
+
+  while ((start = output.find(kLogStr, start)) != std::string::npos) {
+    ++occurrences;
+    start += kLogStr.length();
+  }
+  EXPECT_LT(occurrences, num_iterations);
```
Comment (scv119) [inline src/ray/util/logging_test.cc:100]: oh well. this could be another flaky test.

**Comment by rkooo567** [comment]
We are using spdlog (glog will be removed)

**Review by rkooo567 — APPROVED** [review]
LGTM

**Review by ericl — APPROVED** [review]

**Pull request merged**